src/loaders/index.js

```diff
@@ -171,12 +171,46 @@ export async function createOMETiffLoader({
   const firstImage = await tiff.getImage(0);
   const pool = new Pool();
   const omexmlString = firstImage.fileDirectory.ImageDescription;
+  const metadata = new OMEXML(omexmlString);
+  const dimensions = dimensionsFromOMEXML(metadata);
+  const channelNames = metadata.getChannelNames();
+  const isRgb =
+    metadata.SamplesPerPixel === 3 ||
+    (channelNames.length === 3 && metadata.Type === 'uint8') ||
+    (metadata.SizeC === 3 && channelNames.length === 1 && metadata.Interleaved);
+  const isInterleaved = metadata.Interleaved;
```

Comment:
Could be helpful to wrap this in something so it could be re-used for zarr as well.

Reply:
Mostly copied from the old constructor. This function creates a GeoTIFF object, and derives all other constructor arguments (`firstImage`, `metadata`, `dimensions`) from that source. Then additional fields are parsed in the OMETiff constructor from `metadata` (which itself is derived from the original `tiff`?). I don't see how inheritance is helping us here. 

Metadata are going to be completely different for each class so `getMetadata` isn't going to return the same thing (which is what you want in the first place).